examples/ota-requestor-app/linux/main.cpp

```diff
@@ -155,40 +195,80 @@ void OnQueryImageFailure(void * context, EmberAfStatus status)
 
 void OnConnected(void * context, chip::DeviceProxy * deviceProxy)
```

Comment:
There are now two different times when OTA Requestor would try to establish a CASE session and communciate with a peer node:
1) When sending the `QueryImage` command
2) After receiving `QueryImageResponse`, parse the BDX ImageURI and connect to that provider node ID

Utilizing `OnConnected ` as a way to handle what to do whenever a CASE session has been established. Before this change, `OnConnected` was only for (1) above. Now it has been moved under the case below for `kQueryImage`. Most of the code in `OnQueryImageResponse` has been moved under the case for `kStartBDX` below.